python/pyiceberg/catalog/rest.py

```diff
@@ -225,15 +228,18 @@ def url(self, endpoint: str, prefixed: bool = True, **kwargs) -> str:
         return url + endpoint.format(**kwargs)
 
     def _fetch_access_token(self, credential: str) -> str:
-        client_id, client_secret = credential.split(":")
+        if SEMICOLON in credential:
+            client_id, client_secret = credential.split(SEMICOLON)
+        else:
+            client_id, client_secret = credential, None
```

Comment:
In Java, we do the opposite. If there is no `:`, then the entire credential is used as the client secret.

Reply:
Ah, that was my intention, fixed!